### DISCUSSION TITLE
@Payload Parameter for  Actions

### DISCUSSION THREAD

**Discussion by SiPurdy** [MarkMpn/Sql4Cds#403] [General]
Trying to work out how to do this, but not finding any examples and struggling to find the relevant section of code.

How would somebody execute a Bulk Delete request (https://learn.microsoft.com/en-us/power-apps/developer/data-platform/webapi/reference/bulkdelete?view=dataverse-latest) to create a recurring bulk delete.

Within Sql4Cds it just indicates the existance of a Payload nvarchar(max) parameter (which would make things easy) and I've not found anything on how this should be set, any advice?

Thanks

**Comment by MarkMpn** [comment]
I think you're looking at the `BulkDeleteJob` message rather than `BulkDelete`. The message you need to use is `BulkDelete`, which is the one you've linked to the documentation for, but the parameter types for this message are not supported by SQL 4 CDS. The `BulkDeleteJob` message isn't documented and is marked as private, so it's designed for internal use by Microsoft.

**Reply by SiPurdy** [reply]
Thanks for the quick reply.

My minds broken, it's too cold. I think my mind went down a route that for situations where the input parameters were complex, you just allowed for a json blob to be passed in instead.
